starlark/hashtable.go

```diff
@@ -200,6 +201,51 @@ func (ht *hashtable) lookup(k Value) (v Value, found bool, err error) {
 	return None, false, nil // not found
 }
 
+// count returns the number of elements of iter that are part of ht
+func (ht *hashtable) count(iter Iterator) (int, error) {
+	if ht.table == nil {
+		return 0, nil // empty
+	}
+
+	var k Value
+	count := 0
+	storage := make([][1]big.Word, len(ht.table))
```

Comment:
```
// Use a bitset per table entry to record seen elements of ht.
// Elements are identified by their bucket number and index within the bucket.
// Each bitset gets one word initially, but may grow.
storage := make([]big.Word, len(ht.table))
for ... { bitsets[i].SetBits(storage[i:i+1:i+1]) }
```

Reply:
I will change this, but I'm curious: in memory this will be the same. Isn't it more idiomatic to actually have N arrays of size 1 than having one big array and indexing it?

Also, thanks for the comment, I think it really helps reading this.

Reply:
The code change is minor but this seems like a more direct expression. Either is fine.

Reply:
Done. I was just curious. Better the indexing as that's the style used within the project.